lms/lmsweb/tools/notebook_extractor.py

```diff
@@ -1,10 +1,34 @@
+import difflib
 import re
 from typing import Any, Iterator, Dict, Iterable, Tuple, List
 
-
 Notebook = Dict[str, Any]
 Cell = Dict[str, Any]
-ASSIGNMENT_NAME_REGEX = re.compile(r'Upload\s+\d+', re.IGNORECASE)
+_UPLOAD_KEYWORD = 'Upload'
+ASSIGNMENT_NAME_REGEX = re.compile(fr'{_UPLOAD_KEYWORD}\s+\d+', re.IGNORECASE)
+_MATCHER_THRESHOLD = 0.7
+
+
+def get_similarity(header: str) -> Tuple[str, str]:
+    if not header:
+        return '', ''
+    _header_parts = header.split()
+    if len(_header_parts) < 2:
+        err = (
+            f'invalid "{_UPLOAD_KEYWORD}" - got "{header}" - '
+            f'maybe you forgot the word {_UPLOAD_KEYWORD} or the exercise number?'
+        )
+        return '', err
+    _upload_header = _header_parts[0].lower()
+    s_matcher = difflib.SequenceMatcher(None, _upload_header, _UPLOAD_KEYWORD.lower())
+    typo_msg = (
+        f"You have a typo in {header} - "
```

Comment:
It's better to leave the strings to the frontend. Maybe we should create enum file somewhere(?)